### PR TITLE
Add |prediction_border| metric option to specify decision threshold used to compute some metrics

### PR DESCRIPTION
**Pull request by ivanychev**
The parameter can be specified to some of the metrics for binary classification. Addresses this issue (https://github.com/catboost/catboost/issues/1181)

Usage: `Precision::prediction_border=0.3`

### CODE REVIEW

File: catboost/private/libs/options/metric_options.h [hunk 1]
```diff
@@ -27,6 +27,8 @@ namespace NCatboostOptions {
         TOption<TLossDescription> EvalMetric;
         TOption<TLossDescription> ObjectiveMetric;
         TOption<TVector<TLossDescription>> CustomMetrics;
+
+        static constexpr char PREDICTION_BORDER_PARAM[] = "prediction_border";
```
Comment (nikitxskv) [inline catboost/private/libs/options/metric_options.h]: "prediction_border" -> "proba_border"

File: catboost/libs/metrics/metric.h [hunk 2]
```diff
@@ -343,23 +344,30 @@ THolder<IMetric> MakeRankingAucMetric();
 THolder<IMetric> MakeMultiClassAucMetric(int positiveClass);
 THolder<IMetric> MakeMuAucMetric(const TMaybe<TVector<TVector<double>>>& misclassCostMatrix = Nothing());
 
-THolder<IMetric> MakeBinClassPrecisionMetric(double border = GetDefaultTargetBorder());
+THolder<IMetric> MakeBinClassPrecisionMetric(double targetBorder = GetDefaultTargetBorder(),
```
Comment (nikitxskv) [inline catboost/libs/metrics/metric.h]: Let's remove `targetBorder` parameter from all `Make*Metric` functions.

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@ivanychev, please agree to the CLA (https://github.com/catboost/catboost/blob/master/CONTRIBUTING.md#provide-contributions)

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: [1170556](https://a.yandex-team.ru/review/1170556)

**Comment by ivanychev** [comment]
I hereby agree to the terms of the CLA available at: [https://yandex.ru/legal/cla/?lang=en](https://yandex.ru/legal/cla/?lang=en).

**Comment by arcadia-devtools** [comment]
Check [1170556](https://a.yandex-team.ru/review/1170556) failed: 1 build(s) broken, 84 test(s) broken

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: [1173157](https://a.yandex-team.ru/review/1173157)

**Comment by arcadia-devtools** [comment]
Check [1173157](https://a.yandex-team.ru/review/1173157) failed: 5 build(s) broken, 19 test(s) broken

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: [1176899](https://a.yandex-team.ru/review/1176899)

**Comment by arcadia-devtools** [comment]
Check [1176899](https://a.yandex-team.ru/review/1176899) failed: 28 test(s) broken

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: [1177784](https://a.yandex-team.ru/review/1177784)

**Comment by arcadia-devtools** [comment]
Check [1177784](https://a.yandex-team.ru/review/1177784) failed: 6 test(s) broken

**Comment by nikitxskv** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@nikitxskv, internal review request created: [1178281](https://a.yandex-team.ru/review/1178281)

**Comment by arcadia-devtools** [comment]
Check [1178281](https://a.yandex-team.ru/review/1178281) passed

**Pull request merged**